syntax = "proto3";

package promo;

service PromoService {
    rpc GetProductPromos (ProductPromoRequest) returns (ProductPromoResponse);
    rpc CheckThresholdPromo (ThresholdPromoRequest) returns (ThresholdPromoResponse);
}

message ProductPromoRequest {
    string product_hash = 1;
}

message ProductPromoResponse {
    bool has_discount = 1;
    double final_price = 2;
    bool is_bogo = 3;
    repeated BuyXGetYInfo buy_x_get_y_promos = 4;
    repeated string bundle_product_hashes = 5;
    bool can_purchase_by_points = 6;
    int32 points_required = 7;
}

message BuyXGetYInfo {
    string required_product_hash = 1;
    string discounted_product_hash = 2;
    double discounted_price = 3;
}

message ThresholdPromoRequest {
    double subtotal = 1;
}

message ThresholdPromoResponse {
    bool has_threshold_promo = 1;
    double discount_value = 2;
}
